src/core/js/a11y.js

```diff
@@ -0,0 +1,686 @@
+        _scroll: new Scroll(),
+
+        log: new Log(),
+
+        initialize: function() {
+            this._removeLegacyElements();
+            this.listenToOnce(Adapt, {
+                'app:dataLoaded': this._onDataLoaded,
+                'navigationView:postRender': this._removeLegacyElements
+            }, this);
+            Adapt.on('device:changed', this._setupNoSelect);
+            this.listenTo(Adapt, {
+                'router:location': this._onNavigationStart,
+                'pageView:ready menuView:ready router:plugin': this._onNavigationEnd
+            });
+        },
+
+        _onDataLoaded: function() {
+            this.config = Adapt.config.get('_accessibility');
+            this.config._isActive = false;
+            _.defaults(this.config, this.defaults);
+            Adapt.offlineStorage.set('a11y', false);
+            this.$html.toggleClass('accessibility', this.isEnabled());
+            this._setupNoSelect();
+            this._addFocuserDiv();
+            if (this._isReady) return;
+            this._isReady = true;
+            Adapt.trigger('accessibility:ready');
+        },
+
+        _setupNoSelect: function() {
+            if (!this.config || !this.config._disableTextSelectOnClasses) return;
+            var classes = this.config._disableTextSelectOnClasses.split(' ');
+            var isMatch = false;
+            for (var i = 0, item; item = classes[i++];) {
+                if (this.$html.is(item)) {
+                    isMatch = true;
+                    break;
+                }
+            }
+            this.$html.toggleClass('no-select', isMatch);
+        },
+
+        _addFocuserDiv: function() {
+            if ($('#a11y-focuser').length) return;
+            $('body').append($('<div id="a11y-focuser" class="a11y-ignore" tabindex="-1" role="presentation">&nbsp;</div>'));
+        },
+
+        _removeLegacyElements: function() {
+            var $legacyElements = $('body').children('#accessibility-toggle, #accessibility-instructions');
+            var $navigationElements = $('.navigation').find('#accessibility-toggle, #accessibility-instructions');
+            if (!$legacyElements.length && !$navigationElements.length) return;
+            Adapt.log.warn('REMOVED: #accessibility-toggle and #accessibility-instructions have been removed. Please remove them from all of your .html files.');
+            $legacyElements.remove();
+            $navigationElements.remove();
+        },
+
+        _onNavigationStart: function() {
+            if (!this.isEnabled()) return;
+            // Stop document reading
+            _.defer(function() {
+                Adapt.a11y.toggleHidden('.page, .menu', true);
+            });
+        },
+
+        _onNavigationEnd: function(view) {
+            // Prevent sub-menu items provoking behaviour
+            if (view && view.model && view.model.get('_id') !== Adapt.location._currentId) return;
+            if (!this.isEnabled()) return;
```

Comment:
Minor: since there are two early returns here is it worth combining this with an OR statement in the `if` above?